R/daily_stat.R

```diff
@@ -0,0 +1,134 @@
+daily_stat <- function(data, dt = "date", val = "value", 
+                       by = c("ems_id", "site"), digits = 1,
+                       thresh, n_readings_min,
+                       stat) {
+  data <- group_by_(data, .dots = c(by, dt))
+  data <- 
+    summarise_(data,
+               n_readings = interp(~length(na.omit(value)), 
+                                   value = as.name(val)), 
+               stat       = interp(~fun(value), 
+                                   value = as.name(val), fun = stat), 
+               exceed     = interp(~stat > thresh, thresh = thresh), 
+               valid      = interp(~n_readings >= n_readings_min, 
+                                   n_readings_min = n_readings_min))
+  data$exceed <- ifelse(is.na(data$exceed), FALSE, data$exceed)
+  data$flag <- data$exceed & !data$valid # Flag for data incomplete, but used
+  data$stat <- round(ifelse(data$valid | data$flag, data$stat, NA_real_), digits)
+  data
+}
+
+# Clobbers any date column if there was one.
+pollutant_daily_stat <- function(data, dt, val, by = NULL, pollutant_standard, stat = max_na) {
+  if (!is.null(by)) data <- group_by_(data, .dots = by)
+  data <- mutate_(data, date = interp(~time_to_date(date_time), date_time = as.name(dt))) 
+  daily_stat(data,
+             dt = "date", 
+             val = val, 
+             by = by, 
+             thresh = pollutant_standard, 
+             n_readings_min = 18,
+             stat = stat)
+}
+
+#' Cacluate daily maxima or averages for different sensor readings.
+#'
+#'@param data data frame with date and value
+#'@param dt the name (as a character string) of the date-time column. Default
+#'  \code{"date_time"}
+#'@param val the name (as a character string) of the PM2.5 value column. Default
+#'  \code{"value"}
+#'@param by character vector of grouping variables in data, probably an id if
+#'  using multiple sites. Even if not using multiple sites, you shoud specfify
+#'  the id column so that it is retained in the output.
+#'@param exclude_df a data.frame specifying which date ranges and sites to exclude. 
+#'  The data.frame must have to one date column if specific dates or date-times are to be excluded,
+#'  and exactly two date or date-time columns if a date range is needed.
+#'  must have all the columns of by provided, 
+#'@param exclude_df_dt specifies the one (or two) date columns to be use in the exclude_df data.frame.
+#' @return data frame with the daily averages, can be input into 
+#' @name daily_stat_page
+NULL
+#> NULL
+
+#' @rdname daily_stat_page
+#' @importFrom lubridate is.POSIXt
+#' @export
+o3_daily_max <- function(data, dt = "date_time", val = "rolling8", by = NULL, exclude_df = NULL, exclude_df_dt = NULL) {
+  stopifnot(is.data.frame(data), 
+            is.character(dt),
+            is.character(val),
+            is.POSIXt(data[[dt]]), 
+            is.numeric(data[[val]]))
+  if(!is.null(exclude_df)) data <- exclude_data(data, dt, by, exclude_df, exclude_df_dt)
```

Comment:
Indeed, I incorrectly calculated the data completeness criteria on the dataset with the subsetted data. This should be a relatively easy change to make, I'll try and get this done in the next few hours.